osu.Game/Scoring/ScoreInfoExtensions.cs

```diff
@@ -32,5 +32,13 @@ public static IEnumerable<ScoreInfo> OrderByTotalScore(this IEnumerable<ScoreInf
         /// <param name="score">The <see cref="ScoreInfo"/> to compute the maximum achievable combo for.</param>
         /// <returns>The maximum achievable combo.</returns>
         public static int GetMaximumAchievableCombo(this ScoreInfo score) => score.MaximumStatistics.Where(kvp => kvp.Key.AffectsCombo()).Sum(kvp => kvp.Value);
+
+        /// <summary>
+        /// Retrieves the <see cref="ScoreInfo"/> with the maximum total score.
+        /// </summary>
+        /// <param name="scores">An array of <see cref="ScoreInfo"/>s to retrieve the scoreInfo with maximum total score.</param>
+        /// <returns>The <see cref="ScoreInfo"/> instance with the maximum total score.</returns>
+        public static ScoreInfo? MaxByTopScore(this IEnumerable<ScoreInfo> scores)
+            => scores.MaxBy(info => (info.TotalScore, -info.OnlineID, -info.Date.UtcDateTime.Ticks));
```

Comment:
First of all, if this is a helper now for whatever reason, then it cannot return `ScoreInfo`. Removing the detach means that the `ScoreInfo` instance is now unsafe for off-thread accesses and for wrotes. It should return `Live<ScoreInfo>` instead.

Secondly, why's an optimisation PR changing order, too? Also how does this sorting by anonymous tuple even work?

Reply:
I guess the answer to that last question is [here](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/ValueTuple.cs#L714-L731). TIL.

Reply:
> First of all, if this is a helper now for whatever reason, then it cannot return `ScoreInfo`. Removing the detach means that the `ScoreInfo` instance is now unsafe for off-thread accesses and for writes. It should return `Live<ScoreInfo>` instead. Read [Realm usage rules](https://github.com/ppy/osu/wiki/Realm-usage-rules) for more. (You wouldn't have to bother with `Live<T>` if you just removed the detach in the place it previously was.)

>Note that when subscribing to changes, it is recommended to not locally use Live<T> as the subscription handling ensures that all (update thread) access will be serviced by valid realm instances. That said, if passing the results of a subscription to another class, Live<T> should still be employed.

I think the threading should be handled at the call site instead of being returned by a helper function?
I have updated the code to inline the function since it is only called once and is simple enough.

> Secondly, why's an optimisation PR changing order, too? What are these compare rules, why is `OnlineID` a sort parameter? Also how does this sorting by anonymous tuple even work?

The order is following the original code, but OnlineID is not needed in this case since the data is already filtered.I have updated the code to remove it.

https://github.com/ppy/osu/blob/1868826d692f9d0c0fc13a9184758a1c3518a85f/osu.Game/Scoring/ScoreInfoExtensions.cs#L23-L27

https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.orderby?view=net-6.0
>Sorts the elements of a sequence in ascending order according to a key.

Reply:
> I think the threading should be handled at the call site instead of being returned by a helper function

What you had initially was a general purpose helper that could be called by anything and anyone. Using `Live<T>` is a must in such scenarios.